pkg/recommender/vms/filters.go

```diff
@@ -23,16 +23,44 @@ import (
 type vmFilter func(vm recommender.VirtualMachine, req recommender.ClusterRecommendationReq) bool
 
 // filtersForAttr returns the slice for
-func (s *vmSelector) filtersForAttr(attr string, provider string) ([]vmFilter, error) {
+func (s *vmSelector) filtersForAttr(attr string, provider string, req recommender.ClusterRecommendationReq) ([]vmFilter, error) {
+	var filters []vmFilter
 	// generic filters - not depending on providers and attributes
-	var filters = []vmFilter{s.includesFilter, s.excludesFilter, s.categoryFilter}
+	if req.Includes == nil || len(req.Includes) == 0 {
```

Comment:
invert conditions here, remove the else branches (it makes the code easier to read - eventually the list of registered filters can be logged at the end)

Reply:
thx, fixed